src/e84_geoai_common/llm/core/llm.py

```diff
@@ -87,6 +100,10 @@ class LLMInferenceConfig(BaseModel):
 
     model_config = ConfigDict(strict=True, extra="forbid")
 
+    tools: list[Callable[..., Any]] | None = Field(
```

Comment:
My comment on tools above is mostly from looking at this and having questions around if every tool is always a function and if the tools need invocation schemas, etc.